modules/global/src/de/diedavids/cuba/dataimport/converter/MetaPropertyMatcher.groovy

```diff
@@ -3,24 +3,44 @@ package de.diedavids.cuba.dataimport.converter
 import com.haulmont.chile.core.model.MetaClass
 import com.haulmont.chile.core.model.MetaProperty
 import de.diedavids.cuba.dataimport.entity.AttributeType
+import de.diedavids.cuba.dataimport.entity.ImportAttributeMapper
 import org.apache.commons.lang.StringUtils
 import org.springframework.stereotype.Component
 
 @Component('ddcdi_MetaPropertyMatcher')
 class MetaPropertyMatcher {
 
 
-    String findEntityAttributeForColumn(String column, MetaClass selectedEntity) {
-
-        MetaProperty possibleProperty = findPropertyByColumn(selectedEntity, column)
-
-        String result = ''
-        if (possibleProperty && isSimpleDatatype(possibleProperty)) {
-                result = possibleProperty.name
+    ImportAttributeMapper findEntityAttributeForColumn(Integer index, String column, MetaClass selectedEntity) {
+
+        def inputColumnList = getAttributePath(column)
+        MetaProperty possibleProperty = findPropertyByColumn(selectedEntity, inputColumnList[(0)])
+        def attrType = findAttributeTypeForColumn(inputColumnList[(0)], selectedEntity)
+        def possiblePropertyName =  (attrType) ? possibleProperty?.name : ''
+
+        def result = new ImportAttributeMapper(
+                entityAttribute: possiblePropertyName,
+                attributeType: attrType,
+                fileColumnAlias: column,
+                fileColumnNumber: index,
+        )
+        if (possibleProperty && !isSimpleDatatype(possibleProperty)) {
+            MetaClass assocAttribute = possibleProperty.range?.asClass()
+            if (possiblePropertyName && assocAttribute) {
+                result.associationLookupAttribute = findPropertyByColumn(assocAttribute, inputColumnList[(1)])?.name
```

Comment:
I got a `ArrayIndexOutOfBounds` because inputColumnsList might have only one entry (as in my case).

As well -> I will add some tests around it and nail the features by test cases. Otherwise this bean becomes a beast to maintain...